homeassistant/components/homewizard_energy/sensor.py

```diff
@@ -0,0 +1,217 @@
+)
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import (
+    ATTR_ACTIVE_POWER_L1_W,
+    ATTR_ACTIVE_POWER_L2_W,
+    ATTR_ACTIVE_POWER_L3_W,
+    ATTR_ACTIVE_POWER_W,
+    ATTR_GAS_TIMESTAMP,
+    ATTR_METER_MODEL,
+    ATTR_SMR_VERSION,
+    ATTR_TOTAL_ENERGY_EXPORT_T1_KWH,
+    ATTR_TOTAL_ENERGY_EXPORT_T2_KWH,
+    ATTR_TOTAL_ENERGY_IMPORT_T1_KWH,
+    ATTR_TOTAL_ENERGY_IMPORT_T2_KWH,
+    ATTR_TOTAL_GAS_M3,
+    ATTR_WIFI_SSID,
+    ATTR_WIFI_STRENGTH,
+    CONF_API,
+    CONF_DATA,
+    CONF_DEVICE,
+    COORDINATOR,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+SENSORS: Final[tuple[SensorEntityDescription, ...]] = (
+    SensorEntityDescription(
+        key=ATTR_SMR_VERSION,
+        name="SMR Version",
+        icon="mdi:counter",
+        entity_category=ENTITY_CATEGORY_DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key=ATTR_METER_MODEL,
+        name="Model",
+        icon="mdi:gauge",
+        entity_category=ENTITY_CATEGORY_DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key=ATTR_WIFI_SSID,
+        name="Wifi SSID",
+        icon="mdi:wifi",
+        entity_category=ENTITY_CATEGORY_DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key=ATTR_WIFI_STRENGTH,
+        name="Wifi Strength",
+        icon="mdi:wifi",
+        native_unit_of_measurement=PERCENTAGE,
+        state_class=STATE_CLASS_MEASUREMENT,
+        entity_category=ENTITY_CATEGORY_DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+    ),
+    SensorEntityDescription(
+        key=ATTR_TOTAL_ENERGY_IMPORT_T1_KWH,
+        name="Total Power Import T1",
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        device_class=DEVICE_CLASS_ENERGY,
+        state_class=STATE_CLASS_TOTAL_INCREASING,
+    ),
+    SensorEntityDescription(
+        key=ATTR_TOTAL_ENERGY_IMPORT_T2_KWH,
+        name="Total Power Import T2",
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        device_class=DEVICE_CLASS_ENERGY,
+        state_class=STATE_CLASS_TOTAL_INCREASING,
+    ),
+    SensorEntityDescription(
+        key=ATTR_TOTAL_ENERGY_EXPORT_T1_KWH,
+        name="Total Power Export T1",
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        device_class=DEVICE_CLASS_ENERGY,
+        state_class=STATE_CLASS_TOTAL_INCREASING,
+    ),
+    SensorEntityDescription(
+        key=ATTR_TOTAL_ENERGY_EXPORT_T2_KWH,
+        name="Total Power Export T2",
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        device_class=DEVICE_CLASS_ENERGY,
+        state_class=STATE_CLASS_TOTAL_INCREASING,
+    ),
+    SensorEntityDescription(
+        key=ATTR_ACTIVE_POWER_W,
+        name="Active Power",
+        native_unit_of_measurement=POWER_WATT,
+        device_class=DEVICE_CLASS_POWER,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key=ATTR_ACTIVE_POWER_L1_W,
```

Comment:
I would just inline these constants because these classes itself are constants

Reply:
You mean replace things like `ATTR_ACTIVE_POWER_L1_W` to `"active_power_l1_w"`?

Reply:
Personal preference. These objects are constants and the name is also here. It's not used anywhere else. Now the definition of these objects is unnecessarily spread over 2 files.

Reply:
Just to be sure, I understand that this (-> https://github.com/home-assistant/core/pull/55812/commits/f297592bbc2d11e17ec7830f49cfaf94231cabe8) is what you prefer?

Reply:
👍